scapy/packet.py

```diff
@@ -1238,16 +1238,33 @@ def split_layers(lower, upper, __fval=None, **fval):
 
 
 @conf.commands.register
-def ls(obj=None):
-    """List  available layers, or infos on a given layer"""
-    if obj is None:
+def ls(obj=None, case_sensitive=False):
+    """List  available layers, or infos on a given layer class or name"""
+    is_string = isinstance(obj, basestring)
+
+    if obj is None or is_string:
         
         import __builtin__
-        all = __builtin__.__dict__.copy()
-        all.update(globals())
-        objlst = sorted(conf.layers, key=lambda x: x.__name__)
-        for o in objlst:
-            print "%-10s : %s" %(o.__name__,o.name)
+        all_names = __builtin__.__dict__.copy()
+        all_names.update(globals())
+        all_layers = sorted(conf.layers, key=lambda x: x.__name__)
+
+    if obj is None:
+        for layer in all_layers:
+            print "%-10s : %s" % (layer.__name__, layer.name)
+
+    elif is_string:
```

Comment:
There is no reason: you have a first `if A or B:`, and in that block, a `if A: [...] elif B:`. If `A or B`, and `not A`, then `B`. The `elif B` can safely be replaced by `else`.